contracts/AccountStorage.sol

```diff
@@ -17,6 +17,7 @@ contract AccountStorage {
 
     address[] public allowlist;
     mapping (address => uint256) private indexOf; //1 based indexing. 0 means non-existent
+    mapping (address => bool) private createContractPermissions;
```

Comment:
In terms of storage access, it would probably make more sense to have a single structure that stores all permission flags:
```
struct Account {
  uint32 indexOf;
  bool isAdmin;
  bool canCreateContracts;
}
mapping (address => Account) private accountPermissions;
```
instead of having multiple mappings, one for each permission. When using the struct, we only have a single storage access per user.

I think it would make sense to go even further and share this structure between the different contracts instead of calling several times into the storage contract. This way we can guarantee that we only have a single storage access to read all properties.

Not necessarily in this PR :-)